modules/ROOT/pages/queries/case.adoc

```diff
@@ -0,0 +1,250 @@
+[[case-simple-syntax]]
+=== Syntax 
+
+[source, syntax]
+----
+CASE test
+  WHEN value THEN result
+  [WHEN ...]
+  [ELSE default]
+END
+----
+
+*Arguments:*
+[options="header", cols="1,2"]
+|===
+| Name | Description
+
+| `test`
+| An expression.
+
+| `value`
+| An expression whose result will be compared to `test`.
+
+| `result`
+| The expression returned as output if `value` matches `test`
+
+| `default`
+| The expression to return if no value matches the test expression.
+|===
+
+[[case-simple-examples]]
+=== Example
+
+[source, cypher]
+----
+MATCH (n:Person)
+RETURN
+CASE n.eyes
+  WHEN 'blue'  THEN 1
+  WHEN 'brown' THEN 2
+  ELSE 3
+END AS result, n.eyes
+----
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| result | n.eyes
+| 2      | "brown"
+| 1      | "blue"
+| 3      | "green"
+| 2      | "brown"
+| 1      | "blue"
+2+d|Rows: 5
+|===
+
+[[case-generic]]
+== Generic `CASE`
+
+The generic `CASE` expression supports multiple conditional statements.
+Each row is evaluated in order and without any pre-evaluation until a `true` value is found.
+If no match is found, the expression in the `ELSE` operator is returned.
+If there is no `ELSE` case and no match is found, `null` will be returned.
+
+[[case-generic-syntax]]
+=== Syntax
+
+[source, syntax]
+----
+CASE
+  WHEN predicate THEN result
+  [WHEN ...]
+  [ELSE default]
+END
+----
+
+*Arguments:*
+[options="header", cols="1,2"]
+|===
+| Name | Description
+| `predicate`
+| A predicate is an expression that evaluates to a `BOOLEAN` value.
+In this case, the predicate is tested to find a valid alternative.
+
+| `result`
+| The expression returned as output if `predicate` evaluates to `true`.
+
+| `default`
+| If no match is found, `default` is returned.
+|===
+
+[[case-generic-examples]]
+=== Example
+
+[source, cypher]
+----
+MATCH (n:Person)
+RETURN
+CASE
+  WHEN n.eyes = 'blue' THEN 1
+  WHEN n.age < 40      THEN 2
+  ELSE 3
+END AS result, n.eyes, n.age
+----
+
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+| result | n.eyes  | n.age
+| 2      | "brown" | 38
+| 1      | "blue"  | 25
+| 3      | "green" | 53
+| 3      | "brown" | null
+| 1      | "blue"  | 41
+3+d|Rows: 5
+|===
+
+
+[[expressions-case-null-differentiating]]
+== `CASE` with `null` values, and differentiating simple and generic `CASE` forms
+
+The difference between the two `CASE` expressions boils down to what is evaluated first.
+Two queries matching the `age` property of the `Daniel` node (which is `null`) are used to demonstrate the difference.
+
+In a simple `CASE` expression, the `n.age` is evaluated on the third line in the below query.
+For the node `Daniel` it results in `null`, and this will never match a value test, because `null` cannot equal any other value (including `null` itself).
+The simple `CASE` expression therefore skips the `WHEN` branch, and goes straight to `ELSE`, where, because `null - 10` equals `null`, the resulting age for `Daniel` ten years ago is `null`.
```

Comment:
I find these paragraph descriptions hard to follow. I've made an attempt at breaking down the explanation and relating it to individual lines, what's your thoughts?

![Screenshot from 2023-09-19 11-56-15](https://github.com/neo4j/docs-cypher/assets/114478074/bfaab7b5-8b7d-434f-a369-5fbd03086f39)